Calculator/Calculator/ViewController.swift

```diff
@@ -88,246 +82,50 @@ class ViewController: UIViewController {
             }
         }
         
+        let formatter = NumberFormatter()
         
-        let result = String(operationsArray[0]).components(separatedBy: ".")
-        if Int(result[1]) == 0 {
-            mainLabel.text = result[0]
-        }else{
+        if operationsArray[0].hasSuffix(".0") {
+            mainLabel.text = formatter.string(from: Double(operationsArray[0])! as NSNumber) ?? "n/a"
+        } else {
             mainLabel.text = operationsArray[0]
         }
-          
-        //        for num in some {
-//            if num == "*"{
-//                let num1 = num.index(before: num.endIndex)
-//                print(num1)
-//                result = Int(num.index(before: num)) * Int(num.index(after: num))
-//            }
-//        }
     }
     
-    @IBAction func plusOrLessBotton(_ sender: Any) {
-        if mainLabel.text == "0" {
-            
-            
-        }else if mainLabel.text?.first == "-" {
+    @IBAction func addOrSubtractButton(_ sender: Any) {
+      if mainLabel.text?.first == "-" {
             mainLabel.text?.removeFirst()
-        }
-        else if mainLabel.text?.first != "-" && mainLabel.text?.first != "0" {
+        } else if mainLabel.text?.first != "-" && mainLabel.text?.first != "0" {
             mainLabel.text = "-" + mainLabel.text!
         }
     }
     
     @IBAction func percentage(_ sender: Any) {
-        
-        mainLabel.text = String(Double(mainLabel.text!)! * Double(0.01))
+        mainLabel.text = String((Double(mainLabel.text ?? "0") ?? 0) * Double(0.01))
     }
     
-    @IBAction func divide(_ sender: Any) {
-        if mainLabel.text == "0"{
-            
-            
-        } else if mainLabel.text?.last == "/" || mainLabel.text?.last == "*" || mainLabel.text?.last == "+" || mainLabel.text?.last == "-" {
-            
-        }else{
-            mainLabel.text?.append("/")
-        }
+    @IBAction func operationsButton(_ sender: UIButton) {
+        let characterSet: String = "+-*/0"
```

Comment:
use better names, and here you are using a String and not NSCharacterSet which is mislleading